### PR TITLE
Deadend radical species selection RMG-RMS Twin PR

### PR DESCRIPTION
**Pull request by hwpang**
### Motivation or Problem
During the generation of oligomerization mechanism, I found that current species selection algorithm often prioritize selecting propagation reaction in each iteration, such that RMG has difficulty recognizing the importance of many chain transfer and chain termination reactions. This causes the generated mechanism to contain many radicals that are only consumed by very slow reverse reaction and accumulate to nonphysical concentrations in the longer time span simulation. The algorithm designs to help RMG recognize these important but low flux chain transfer and chain termination reactions.

### Description of Change
I add tolerance to move reaction to core based on the dead-end radical algorithm. This is a RMG-RMS twin PR.

### CODE REVIEW

**Comment by rwest** [comment]
Please can you add documentation and example usage?

File: documentation/source/users/rmg/input.rst [hunk 1]
```diff
@@ -469,6 +469,22 @@ For example ::
 			  branchingRatioMax=1.0,
 		)
 
+Advanced Setting: Deadend Radical Elimination (RMS Reactors only)
+-----------------------------------------------------------------
+There are many important low-flux pathways that consume radicals, such as chain transfer and chain termination reactions in polymerization.
```
Comment (mjohnson541) [inline documentation/source/users/rmg/input.rst:474]: My understanding is that the deadend radical algorithm solves the problem that in molecular growth cases with many radicals and fast propagation pathways, non-propagation reactions that consume radicals individually can be so low flux that they do not affect radical concentrations and thus do not register in the flux or branching algorithm, but together control the overall radical loss and thus are very important to include? 

I don't think the use case for the algorithm is made clear here.

**Review by mjohnson541 — CHANGES_REQUESTED** [review]

File: documentation/source/users/rmg/input.rst [hunk 2]
```diff
@@ -471,12 +471,14 @@ For example ::
 
 Advanced Setting: Deadend Radical Elimination (RMS Reactors only)
 -----------------------------------------------------------------
-There are many important low-flux pathways that consume radicals, such as chain transfer and chain termination reactions in polymerization.
-Each of these reactions is often low-flux, but the overall flux is non-negligible, since it controls the radical concentration.
-These pathways are difficult for flux criterion or branching criterion to identify, which lead to the existence of deadend radicals in the
-generated mechanism, i.e., radicals that are only consumed by very slow reverse reactions.
-Deadend radical elimination feature is designed to identify important chain transfer and termination edge reactions based on 
-their contribution to radical consumption and termination.
```
Comment (mjohnson541) [inline documentation/source/users/rmg/input.rst]: Maybe: 
```
When generating mechanisms involving significant molecular growth (such as in polymerization) there are many possible radicals and fast propagation pathways compete with very slow termination and chain transfer reactions. These slow reactions individually usually have a negligible impact on the model and thus will not be picked up by the flux or branching criteria. However, together they can have a very significant impact on the overall radical concentration and need included in the generated mechanism. The deadend radical elimination algorithm identifies important chain transfer and termination reactions in the edge based on their flux ratio with radical consumption and termination reactions in the core. 
```

Comment (hwpang) [reply documentation/source/users/rmg/input.rst]: I like it! I just made the changes

**Comment by hwpang** [comment]
I updated the documentation with your suggestion and rebased with main!

File: documentation/source/users/rmg/input.rst [hunk 3]
```diff
@@ -469,6 +469,21 @@ For example ::
 			  branchingRatioMax=1.0,
 		)
 
+Advanced Setting: Deadend Radical Elimination (RMS Reactors only)
+-----------------------------------------------------------------
+When generating mechanisms involving significant molecular growth (such as in polymerization),
+there are many possible radicals and fast propagation pathways competing with very slow termination and chain transfer reactions.
```
Comment (mjohnson541) [inline documentation/source/users/rmg/input.rst:475]: This should be "compete" not "competing"

Comment (rwest) [reply documentation/source/users/rmg/input.rst:475]: Disagree. 
There are reactions competing... 
Reactions compete...

Comment (rwest) [reply documentation/source/users/rmg/input.rst:475]: I see the issue now. More complex fix. What is the "and" joining? _Radicals_ and _reactions, or _are_ and _compete_. Once you have decided that, you can restructure the sentence to make it clearer.

Comment (rwest) [reply documentation/source/users/rmg/input.rst:475]: Change the "and" to ", so" and "competing" to "compete"

Comment (hwpang) [reply documentation/source/users/rmg/input.rst]: @rwest Thanks for the in-person discussion today! I really like your version and I change the documentation accordingly.

**Comment by hwpang** [comment]
I rebased the PR

**Review by mjohnson541 — APPROVED** [review]

**Pull request merged**